sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -102,7 +102,7 @@ class SessionCatalog(
   @GuardedBy("this")
   protected var currentDb: String = formatDatabaseName(DEFAULT_DATABASE)
 
-  private val validNameFormat = "([\\w_]+)".r
+  private val validNameFormat = "([a-zA-Z0-9]+[\\w]*)".r
```

Comment:
can we use `0-9` as the first letter of table name?

Reply:
yes, same as before.. just restricts name not to start with underscore

Reply:
we have restricted only _ by this change ?

Reply:
and abiding by hive standard